modules/azure/servicebus.go

```diff
@@ -0,0 +1,383 @@
+			return nil, err
+		}
+	}
+
+	return &results, nil
+}
+
+// ListServiceBusNamespaceNames list names of all SB namespaces in all resource groups in the given subscription ID.
+func ListServiceBusNamespaceNames(t *testing.T, subscriptionID string) *[]string {
+	results, err := ListServiceBusNamespaceNamesE(subscriptionID)
+	
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListServiceBusNamespaceIDsE list IDs of all SB namespaces in all resource groups in the given subscription ID. This function would fail the test if there is an error.
+func ListServiceBusNamespaceIDsE(subscriptionID string) (*[]string, error) {
+	sbNamespace, err := ListServiceBusNamespaceE(subscriptionID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]string, 0)
+	for _, namespace := range *sbNamespace {
+		results = append(results, *namespace.ID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &results, nil
+}
+
+// ListServiceBusNamespaceIDs list IDs of all SB namespaces in all resource groups in the given subscription ID.
+func ListServiceBusNamespaceIDs(t *testing.T, subscriptionID string) *[]string {
+	results, err := ListServiceBusNamespaceIDsE(subscriptionID)
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListServiceBusNamespaceByResourceGroupE list all SB namespaces in the given resource group. This function would fail the test if there is an error.
+func ListServiceBusNamespaceByResourceGroupE(subscriptionID string, resourceGroup string) (*[]servicebus.SBNamespace, error) {
+	nsClient, err := serviceBusNamespaceClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	iteratorSBNamespace, err := nsClient.ListByResourceGroupComplete(context.Background(), resourceGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]servicebus.SBNamespace, 0)
+
+	for iteratorSBNamespace.NotDone() {
+		results = append(results, iteratorSBNamespace.Value())
+		err = iteratorSBNamespace.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &results, nil
+}
+
+// ListServiceBusNamespaceByResourceGroup list all SB namespaces in the given resource group.
+func ListServiceBusNamespaceByResourceGroup(t *testing.T, subscriptionID string, resourceGroup string) *[]servicebus.SBNamespace {
+	results, err := ListServiceBusNamespaceByResourceGroupE(subscriptionID, resourceGroup)
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListServiceBusNamespaceNamesByResourceGroupE list names of all SB namespaces in the given resource group. This function would fail the test if there is an error.
+func ListServiceBusNamespaceNamesByResourceGroupE(subscriptionID string, resourceGroup string) (*[]string, error) {
+	sbNamespace, err := ListServiceBusNamespaceByResourceGroupE(subscriptionID, resourceGroup)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]string, 0)
+	for _, namespace := range *sbNamespace {
+		results = append(results, *namespace.Name)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
NIT: this code block is repeated between `ListServiceBusNamespaceNamesE` and `ListServiceBusNamespaceNamesByResourceGroupE`. Can you create an internal helper that plucks the namespace names given a slice of namespaces?

This comment applies also to `ID` attribute flavor.